Sources/DeepLearning/Layer.swift

```diff
@@ -367,3 +367,36 @@ public struct Dropout<Scalar>: Layer
         }
     }
 }
+
+@_fixed_layout
+public struct UpSampling2D<Scalar>: Layer
+    where Scalar: BinaryFloatingPoint & Differentiable & TensorFlowScalar {
+
+    @noDerivative public let size: Int32
+
+    public init(size: Int32) {
+        self.size = size
+    }
+
+    @differentiable(wrt: (self, input))
+    public func applied(to input: Tensor<Scalar>) -> Tensor<Scalar> {
+        let batchSize = input.shape[0]
+        let height = input.shape[1]
+        let width = input.shape[2]
+        let channels = input.shape[3]
+        let reshapeSize = Tensor<Int32>([batchSize,
```

Comment:
One minor issue: You don't have to break lines for any of these statements, as long as everything fits within 100 columns.

Reply:
Done.

Please let me know if there're any more issues I need to fix.

Reply:
Sorry, i wasn't super clear -- i was referring to the lines that are broken for the array literal, not the `let`s.

I actually liked temporary variables for dimension names :)

Reply:
Also, you can bring the right multiplication operand `scaleOnes` to its previous line.

Reply:
Ah, alright, thanks. Fixed.